Dockerfile

```diff
@@ -88,8 +88,8 @@ RUN \
 ARG HLS_VERSION=1.8.0.0
 RUN \
   set -o errexit -o xtrace; \
-  if test -n "$HLS_VERSION"; then \
-  ghcup install hls "$HLS_VERSION" --set; \
+  if test -n "$HLS_VERSION" && "$GHC_VERSION" == "9.2.5"; then \
```

Comment:
I can come back and make a conditional here to install 1.8 for 9.0.2 if we really want, but I'd prefer to get things working end to end in smurf first.

Reply:
Here is what I'd like:

- If configured to do so, install HLS from source.
- Otherwise attempt to install HLS via GHCup.
- Or do nothing.

I feel like the easiest way to support that would be to have two environment variables: `HLS_GIT_REF` and `HLS_VERSION`. If `HLS_GIT_REF` is set, HLS will be compiled from source using `ghcup compile hls -g $HLS_GIT_REF`. Otherwise, if `HLS_VERSION` is set, HLS will be installed using `ghcup install hls $HLS_VERSION`. If neither are set, HLS is not installed. 

Or perhaps we could try to be fancy by keeping the single `HLS_VERSION` variable and checking to see if it looks like a commit. If it does, compile; otherwise install. 

``` sh
if echo "$HLS_VERSION" | grep --extended-regexp --quiet '^[0-9a-f]{40}$'
then echo compile
else echo install
fi
```

For example:

``` sh
$ if echo '1.8.0.0' | egrep -q '^[0-9a-f]{40}$'; then echo compile; else echo install; fi
install
```

``` sh
$ if echo '5d56aa70a84807d7659e72eacd4d91fee08dbdbb' | egrep -q '^[0-9a-f]{40}$'; then echo compile; else echo install; fi
compile
```

That way we can use `HLS_VERSION=1.8.0.0` for `GHC_VERSION=9.0.2` (as we currently do), but also support `HLS_VERSION=5d56aa70a84807d7659e72eacd4d91fee08dbdbb` for `GHC_VERSION=9.2.5` and `GHC_VERSION=9.4.3`. 

What do you think?